### PR TITLE
remove BASHisms

### LINKED ISSUE #1264: 'nosetests' fail on Mac OS X
with latest master (94d679f2e), Python 2.7.2,

```13.4.0 Darwin Kernel Version 13.4.0: Wed Mar 18 16:20:14 PDT 2015; root:xnu-2422.115.14~1/RELEASE_X86_64 x86_64.```

I get:

```
======================================================================
FAIL: tests.test_read_parsers.test_read_pair_iterator_in_error_mode_xfail
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/t/dev/ipy/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/t/dev/khmer/tests/test_read_parsers.py", line 338, in test_read_pair_iterator_in_error_mode_xfail
    assert "Invalid read pair" in str(exc), str(exc)
AssertionError

----------------------------------------------------------------------
Ran 349 tests in 4.954s
```

### PR DESCRIPTION
**Pull request by mr-c**
Might fix #1264 #1263

### CODE REVIEW

**Comment by mr-c** [comment]
@ctb can you test this?

File: ChangeLog [hunk 1]
```diff
@@ -1,3 +1,11 @@
+2015-08-17  Michael R. Crusoe  <crusoe@ucdavis.edu>
+
+   * Makefile: remove BASH shell designation that appears to be incompatible
+   with OS X Mavericks & virtualenv; refactored out BASH-isms for those who's
```
Comment (ctb) [inline ChangeLog]: whose

File: ChangeLog [hunk 2]
```diff
@@ -1,3 +1,11 @@
...
+   default shell isn't BASH (Debian, and others).
+   * lib/test-read-alinger.cc,read_aligner_test.sh: removed unused test files
```
Comment (ctb) [inline ChangeLog]: test-read-aligner

**Comment by ctb** [comment]
Yes, it fixes things.

**Comment by mr-c** [comment]
Sorry for the goofed PR. Updated with what was promised in the ChangeLog but didn't actually make it the first round.

**Comment by mr-c** [comment]
- [ ] Is it mergeable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?
- [ ] Do the changes respect streaming IO? (Are they
  tested for streaming IO?)
- [ ] Is the Copyright year up to date?

**Comment by mr-c** [comment]
@ctb @camillescott Ready for merge

**Comment by camillescott** [comment]
*shrug* LGTM

**Comment by mr-c** [comment]
Thank-ee!

On Wed, Aug 19, 2015 at 6:52 PM Camille Scott <notifications@github.com>
wrote:

> *shrug* LGTM
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dib-lab/khmer/pull/1266#issuecomment-132848911>.
>
-- 
Michael R. Crusoe: Programmer & Bioinformatician crusoe@ucdavis.edu
The lab for Data Intensive Biology; University of California, Davis
https://impactstory.org/MichaelRCrusoe http://twitter.com/biocrusoe

**Pull request merged**